src/functional/iterator/embedded.py

```diff
@@ -245,18 +304,20 @@ def group_offsets(*offsets):
     return complete_offsets, tag_stack
 
 
-def shift_position(pos, *complete_offsets, offset_provider):
+def shift_position(
+    pos: MaybePosT, *complete_offsets: tuple[str, int], offset_provider: OffsetProviderT
+) -> MaybePosT:
     if pos is None:
         return None
-    new_pos = pos.copy()
+    new_pos: MaybePosT = pos.copy()
```

Comment:
```suggestion
    new_pos: Pos = pos.copy()
```

Reply:
Yes, I wanted to fix that, but just like this mypy doesn't like it. Maybe we can have a look together.